### ISSUE TITLE
Output of Elmo changes all the time

### ISSUE THREAD

**Issue by weiguowilliam** [allenai/allennlp#3845]
**System (please complete the following information):**
 - OS: OSX
 - Python version: 3.6
 - AllenNLP version: newest version from pip
 - PyTorch version: 
**Question**
The output of ElmoEmbedder() changes. The printed vector will change slightly during the loop.
How can I stablize it?
Thank you.

```
elmo = ElmoEmbedder()

for i in range(2):
    question = "Guys we're witnessing the Karl Malone and John Stockton of League. Our children will be so jelly when we tell them stories of these legends."
    nlp = spacy.load('en_core_web_sm')
    tmp_q_tokens = nlp(question) 
    q_tokens = [i.text for i in tmp_q_tokens]
    q_embeddings = elmo.embed_sentence(q_tokens)

    exact_idx = q_tokens.index('John')
    print(q_tokens)
    print(q_embeddings[:,8].sum(axis=0))
```

**Comment by ZhaofengWu** [comment]
See https://github.com/allenai/allennlp/blob/master/docs/tutorials/how_to/elmo.md#notes-on-statefulness-and-non-determinism

**Comment by weiguowilliam** [comment]
> See https://github.com/allenai/allennlp/blob/master/docs/tutorials/how_to/elmo.md#notes-on-statefulness-and-non-determinism

Thank you for your reply. I have read this and several related issues. But it seems the nondeterminism is still not solved yet, right?

**Comment by weiguowilliam** [comment]
> Yes `ElmoEmbedder` is still non-deterministic. That's just how ELMo works.

Thank you for your reply. So if I want to generate vector representation for a word, is it better if I generate multiple times and take the average vector?

**Comment by schmmd** [comment]
@weiguowilliam I don't think you need to average results, but it's best if you warm up the model with a number of examples first.
